python/pyspark/sql/column.py

```diff
@@ -250,11 +250,39 @@ def __iter__(self):
         raise TypeError("Column is not iterable")
 
     # string methods
+    _rlike_doc = """ Return a Boolean :class:`Column` based on a regex match.\n
+                :param other: an extended regex expression\n
+
+                >>> df.filter( df.name.rlike('ice$') ).collect()
+                [Row(name=u'Alice', age=1)]
+                """
+    _like_doc = """ Return a Boolean :class:`Column` based on a SQL LIKE match.\n
+               :param other: a SQL LIKE pattern\n
+               See :func:`pyspark.sql.Column.rlike` for a regex version\n
+
+               >>> df.filter( df.name.like('Al%') ).collect()
+               [Row(name=u'Alice', age=1)]
+                """
+    _startswith_doc = ''' Return a Boolean :class:`Column` based on a string match.\n
```

Comment:
This documentation is different than the Scala docs, we generally try and keep them the same in other places so it might make sense to update both.